modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -826,11 +828,8 @@ class TaskProcessor {
             }
 
             final lock = lockManager.acquire(hash)
-            final workDir = task.getWorkDirFor(hash)
             try {
-                if( resumeDir != workDir )
-                    exists = workDir.exists()
-                if( !exists && !workDir.mkdirs() )
+                if( !workDir.mkdirs() )
```

Comment:
I've been wondering about this lock over the task directory. I think the purpose is to prevent two tasks from using the same directory. But in that case maybe it should be over the previous try-catch block?

It should prevent two tasks from checking the same directory at the same time, because that is how a task determines whether to use the directory or try a different one

Reply:
It may. however what I'm thinking that to solve this issue it should be assumed a new task run should *always* use a newly created directory. Not sure this logic satisfy it.